manifests/params.pp

```diff
@@ -272,7 +272,8 @@
       $osreleasemajor = regsubst($::operatingsystemrelease, '^(\d+)\..*$', '\1') # workaround for the possibly missing operatingsystemmajrelease
       $agent_restart_command = $osreleasemajor ? {
         '6'     => "/sbin/service ${service_name} reload",
-        '7'     => "/usr/bin/systemctl reload-or-restart ${service_name}",
+        #'7'     => "/usr/bin/systemctl reload-or-restart ${service_name}",
+        '7'     => "/bin/kill -HUP $(pgrep -f '/usr/bin/ruby /usr/bin/puppet agent')",
```

Comment:
I'm not sure if it works and I don't have a CentOS 7 machine with puppet ready, but how about ```systemctl kill puppet --signal HUP```?